core/observe.c

```diff
@@ -843,6 +843,160 @@ void observe_step(lwm2m_context_t * contextP,
+    lwm2m_data_free(size, dataP);
+    if (ret < 0)
+    {
+        return COAP_500_INTERNAL_SERVER_ERROR;
+    }
+    else
+    {
+        length = ret;
+    }
+
+    if (shortServerID == 0
+     && contextP->serverList != NULL
+     && contextP->serverList->next == NULL)
+    {
+        // Only 1 server
+        shortServerID = contextP->serverList->shortID;
+    }
+
+    ret = COAP_404_NOT_FOUND;
+    for (targetP = contextP->serverList; targetP != NULL; targetP = targetP->next)
+    {
+        bool mute = true;
+        lwm2m_data_t *muteDataP;
+        int muteSize;
+
+        if (shortServerID != 0 && shortServerID != targetP->shortID) continue;
+        if (targetP->sessionH == NULL
+         || (targetP->status != STATE_REGISTERED
+          && targetP->status != STATE_REG_UPDATE_PENDING
+          && targetP->status != STATE_REG_UPDATE_NEEDED
+          && targetP->status != STATE_REG_FULL_UPDATE_NEEDED))
+        {
+            if (ret == COAP_404_NOT_FOUND) ret = COAP_405_METHOD_NOT_ALLOWED;
+            if (shortServerID == 0) continue;
+            break;
+        }
+
+        LWM2M_URI_RESET(&uri);
+        uri.objectId = LWM2M_SERVER_OBJECT_ID;
+        uri.instanceId = targetP->servObjInstID;
+        uri.resourceId = LWM2M_SERVER_MUTE_SEND_ID;
+        if (COAP_205_CONTENT == object_readData(contextP, &uri, &muteSize, &muteDataP))
+        {
+            lwm2m_data_decode_bool(muteDataP, &mute);
+            lwm2m_data_free(muteSize, muteDataP);
+        }
+        if (mute)
+        {
+            if (ret == COAP_404_NOT_FOUND) ret = COAP_405_METHOD_NOT_ALLOWED;
+            if (shortServerID == 0) continue;
+            break;
+        }
+
+        LWM2M_URI_RESET(&uri);
+        transactionP = transaction_new(targetP->sessionH, COAP_POST, NULL, &uri, contextP->nextMID++, 0, NULL);
+        if (transactionP == NULL)
+        {
+            if (shortServerID == 0) lwm2m_free(buffer);
+            ret = COAP_500_INTERNAL_SERVER_ERROR;
+            // Going to the next server likely won't fix this, just get out.
+            break;
+        }
+
+        coap_set_header_uri_path(transactionP->message, "/"URI_SEND_SEGMENT);
+        coap_set_header_content_type(transactionP->message, format);
+        coap_set_payload(transactionP->message, buffer, length);
+
+        transactionP->callback = callback;
+        transactionP->userData = userData;
+
+        contextP->transactionList = (lwm2m_transaction_t *)LWM2M_LIST_ADD(contextP->transactionList, transactionP);
+
+        ret = transaction_send(contextP, transactionP);
+        if (ret == NO_ERROR)
+        {
+            oneGood = true;
+        }
+        else
+        {
+            LOG_ARG("transaction_send failed for %d: 0x%02X!", targetP->shortID, ret);
+        }
+        if (shortServerID != 0) break;
+    }
+    if (buffer)
+    {
+        lwm2m_free(buffer);
+        coap_set_payload(transactionP->message, NULL, 0);
```

Comment:
I get a Segmentation Fault inside the `coap_set_payload` function when trying to send something from a client to a server.

Reproduce:
* Start the example server provided with Wakaama (`lwm2mserver`)
* Start the example client provided with Wakaama (`lwm2mclient`)
* Registration of client is successful
* Try to send an update from the client cli: `send 123 /31024/10`

Reply:
I think I know what caused the segmentation fault.
I tested the "send command" only with [Leshan demo server](https://leshan.eclipseprojects.io/) and [Friendly LwM2M server](https://friendly-tech.com/products/lwm2m-client-iot/), I haven't tested it with Wakaama server, so I don't know if Wakaama server can handle these commands.

In any case, my intention here was to clean up the `transaction->message->payload` pointer, but I have not taken into account that it may happen that the `transactionP` has not been allocated, in your specific case I think it is due to the fact that `if (mute)`  condition is `true`:
https://github.com/eclipse/wakaama/blob/02d94f75fc451a50e43ed025907821cf7f87f68c/core/observe.c#L943-L948

causing early exit from the "for cycle", thus causing the non-allocation of the `transactionP`, I'll fix this with the next PR.

Also I just now noticed that `mute` is `true` because the resource `/1/0/23` was not implemented in the `examples/object_server.c`
 `/1/0/23` resource was added with LwM2M v1.1 and is used to enable/disable the LwM2M Client Send command capability, I will add this resource with the next PR.